source/src/microapp/cs_MicroappController.cpp

```diff
@@ -253,52 +253,54 @@ uint8_t* MicroappController::getOutputMicroappBuffer() {
 }
 
 void MicroappController::setOperatingState(uint8_t appIndex, MicroappOperatingState state) {
+	LOGd("setOperatingState appIndex=%u state=%u", appIndex, state);
 	if (appIndex > 0) {
 		LOGi("Multiple apps not supported yet");
 		return;
 	}
 	uint8_t runFlag = (state == MicroappOperatingState::CS_MICROAPP_RUNNING) ? 1 : 0;
 	bluenet_ipc_data_t ipcData;
+
+	// We can just overwrite all, as a newer IPC version will be written and read by bluenet only.
 	ipcData.bluenetRebootData.ipcDataMajor               = BLUENET_IPC_BLUENET_REBOOT_DATA_MAJOR;
 	ipcData.bluenetRebootData.ipcDataMinor               = BLUENET_IPC_BLUENET_REBOOT_DATA_MINOR;
+
+	memset(ipcData.bluenetRebootData.microapp, 0, sizeof(ipcData.bluenetRebootData.microapp));
 	ipcData.bluenetRebootData.microapp[appIndex].running = runFlag;
 
-	// TODO: this does not handle IPC minor updates well.
-	IpcRetCode ipcCode = setRamData(IPC_INDEX_MICROAPP_STATE, ipcData.raw, sizeof(bluenet_ipc_bluenet_data_t));
+	IpcRetCode ipcCode = setRamData(IPC_INDEX_MICROAPP_STATE, ipcData.raw, sizeof(ipcData.bluenetRebootData));
 	if (ipcCode != IPC_RET_SUCCESS) {
-		LOGi("Error in setting IPC ram data: %i", ipcCode);
+		LOGw("Failed to set IPC data: ipcCode=%i", ipcCode);
 	}
 }
 
 MicroappOperatingState MicroappController::getOperatingState(uint8_t appIndex) {
+	LOGd("getOperatingState appIndex=%u", appIndex);
 	MicroappOperatingState state = MicroappOperatingState::CS_MICROAPP_NOT_RUNNING;
-	if (!isRamDataPresent(IPC_INDEX_MICROAPP_STATE)) {
```

Comment:
If all calls to this function isRamDataPresent are removed, remove the function itself too. Perhaps add some comments to cs_IpcRamData.h that it's hard to find out if data is present because RAM doesn't need to be initialized at all zeros. Hence, the only thing we have is, something where we know that data is present because the checksum is correct. If the checksum is incorrect all bets are off. We can't trust the major or minor to be the same w.r.t. the current software because the version can be bumped. We can't trust the position of all the other fields because they can be shifted around when the major is bumped. The function `isRamDataPresent` should have been read as `isRamDataLikelyPresent` and the negation was only covering the situation in which the data was properly cleared before and subsequently not written. This was the reason why we did a clear of this data after a new install. Perhaps we should write this down somewhere.

Reply:
Written it down at the top of the header.